app/src/main/java/com/dodo/flashcards/presentation/MainActivity.kt

```diff
@@ -11,39 +12,60 @@ import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavHostController
+import com.dodo.flashcards.architecture.Router
 import com.dodo.flashcards.presentation.register_screen.RegisterScreen
 import com.dodo.flashcards.presentation.theme.FlashcardsAppTheme
+import com.dodo.flashcards.util.Screen.*
 import dagger.hilt.android.AndroidEntryPoint
+import com.dodo.flashcards.presentation.MainDestination.*
+import com.google.accompanist.navigation.animation.rememberAnimatedNavController
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
 
 @AndroidEntryPoint
-class MainActivity : ComponentActivity() {
+class MainActivity : ComponentActivity(), Router<MainDestination> {
+
+    lateinit var navController: NavHostController
+    private lateinit var auth: FirebaseAuth
```

Comment:
Since we don't have MainViewModel yet this works, in the future we'll move this logic all there and inject FirebaseAuth there